src/naive/date.rs

```diff
@@ -643,53 +640,28 @@ impl NaiveDate {
             return Some(self);
         }
 
-        // Copy `i32::MIN` here so we don't have to do a complicated cast
-        match months.0 <= 2_147_483_648 {
-            true => self.diff_months(-(months.0 as i32)),
-            false => None,
-        }
+        self.diff_months(months, false)
     }
 
-    fn diff_months(self, months: i32) -> Option<Self> {
-        let (years, left) = ((months / 12), (months % 12));
-
-        // Determine new year (without taking months into account for now
-
-        let year = if (years > 0 && years > (MAX_YEAR - self.year()))
-            || (years < 0 && years < (MIN_YEAR - self.year()))
-        {
-            return None;
-        } else {
-            self.year() + years
-        };
-
-        // Determine new month
-
-        let month = self.month() as i32 + left;
-        let (year, month) = if month <= 0 {
-            if year == MIN_YEAR {
-                return None;
-            }
-
-            (year - 1, month + 12)
-        } else if month > 12 {
-            if year == MAX_YEAR {
-                return None;
-            }
+    fn diff_months(self, months: Months, forwards: bool) -> Option<Self> {
+        // safe as .month() returns [1..12]
+        // can be further improved when .month() returns a Month
+        let month: crate::Month = num_traits::FromPrimitive::from_u32(self.month()).unwrap();
```

Comment:
I don't see the point of using `num_traits::FromPrimitive` and using `unwrap()` if we can just use `as` here.

Reply:
I need the actual `Month` type here as that's what `overflowing_add/sub` take as the `&self` parameter. This will be able to be cleaned up later if/when we fix #727

Reply:
I don't think it is possible to go from `u32` to an enumeration thought?